substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrChunkWriter.java

```diff
@@ -231,40 +257,36 @@ private void writeRepositories(JfrRepository[] constantPools) throws IOException
         }
     }
 
-    private long writeMetadataEvent(byte[] metadataDescriptor) throws IOException {
-        long start = beginEvent();
+    private SignedWord writeMetadataEvent(byte[] metadataDescriptor) throws IOException {
+        SignedWord start = beginEvent();
         writeCompressedLong(JfrEvents.MetadataEvent.getId());
         writeCompressedLong(JfrTicks.elapsedTicks());
         writeCompressedLong(0); // duration
         writeCompressedLong(0); // metadata id
-        file.write(metadataDescriptor); // payload
+        writeBytes(metadataDescriptor); // payload
         endEvent(start);
         return start;
     }
 
     public boolean shouldRotateDisk() {
         assert lock.isHeldByCurrentThread();
-        try {
-            return file != null && file.length() > notificationThreshold;
-        } catch (IOException ex) {
-            Logger.log(JFR_SYSTEM, ERROR, "Could not check file size to determine chunk rotation: " + ex.getMessage());
-            return false;
-        }
+        return filename != null && getFileSupport().size(fd).greaterThan(WordFactory.signed(notificationThreshold));
```

Comment:
This should check if the file descriptor is valid (instead of checking for the filename).